internal/api/api.go

```diff
@@ -189,10 +315,10 @@ type jsonRPCResponseBody struct {
 }
 
 func call(
-	ctx context.Context, client *http.Client,
+	ctx context.Context, client *http.Client, timeout time.Duration,
 	remoteUrl, token, rpc, method string, params []any,
 ) (json.RawMessage, error) {
-	innerCtx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	innerCtx, cancel := context.WithTimeout(ctx, timeout)
```

Comment:
@Foxboron I'm exactly using timeouts with the usage of contexts. Maybe I'm missing something from your previous comment 🤔